pep-0572.rst

```diff
@@ -426,102 +574,43 @@ Once find() returns -1, the loop terminates. If ``:=`` binds as loosely as
 While this behaviour would be convenient in many situations, it is also harder
 to explain than "the := operator behaves just like the assignment statement",
 and as such, the precedence for ``:=`` has been made as close as possible to
-that of ``=``.
+that of ``=`` (with the exception that it binds tighter than comma).
 
 
-Migration path
-==============
-
-The semantic changes to list/set/dict comprehensions, and more so to generator
-expressions, may potentially require migration of code. In many cases, the
-changes simply make legal what used to raise an exception, but there are some
-edge cases that were previously legal and now are not, and a few corner cases
-with altered semantics.
-
-
-The Outermost Iterable
-----------------------
-
-As of Python 3.7, the outermost iterable in a comprehension is special: it is
-evaluated in the surrounding context, instead of inside the comprehension.
-Thus it is permitted to contain a ``yield`` expression, to use a name also
-used elsewhere, and to reference names from class scope. Also, in a genexp,
-the outermost iterable is pre-evaluated, but the rest of the code is not
-touched until the genexp is first iterated over. Class scope is now handled
-more generally (see above), but if other changes require the old behaviour,
-the iterable must be explicitly elevated from the comprehension::
-
-    # Python 3.7
-    def f(x):
-        return [x for x in x if x]
-    def g():
-        return [x for x in [(yield 1)]]
-    # With PEP 572
-    def f(x):
-        return [y for y in x if y]
-    def g():
-        sent_item = (yield 1)
-        return [x for x in [sent_item]]
-
-This more clearly shows that it is g(), not the comprehension, which is able
-to yield values (and is thus a generator function). The entire comprehension
-is consistently in a single scope.
-
-The following expressions would, in Python 3.7, raise exceptions immediately.
-With the removal of the outermost iterable's special casing, they are now
-equivalent to the most obvious longhand form::
-
-    gen = (x for x in rage(10)) # NameError
-    gen = (x for x in 10) # TypeError (not iterable)
-    gen = (x for x in range(1/0)) # ZeroDivisionError
-
-    def <genexp>():
-        for x in rage(10):
-            yield x
-    gen = <genexp>() # No exception yet
-    tng = next(gen) # NameError
-
-
-Open questions
-==============
+Allowing commas to the right
+----------------------------
 
-Importing names into comprehensions
------------------------------------
+Some critics have claimed that the assignment expressions should allow
+unparenthesized tuples on the right, so that these two would be equivalent::
 
-A list comprehension can use and update local names, and they will retain
-their values from one iteration to another. It would be convenient to use
-this feature to create rolling or self-effecting data streams::
+    point = x, y
+    (point := x, y)
```

Comment:
The outer parens don't really clarify much; perhaps `point := (x, y)` would be more indicative?

Reply:
Ah, you're right. These critics want the following to be equivalent:
```
(point := (x, y))
(point := x, y)
```
The rest of this section is my refutation of that position. I'll fix it.